### PR TITLE
Remove duplicate types In Area chart

### PR DESCRIPTION
**Pull request by HHongSeungWoo**
## Description

~~In the stack area chart, when `connectNulls` is set to true and there's a null value, I have adjusted it so that the graph now smoothly connects without any breaks~~

I removed duplicate types and did some code refactoring.

## Related Issue

~~#2176, #1601, maybe #3358~~

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes
- [x] All new and existing tests passed.

### CODE REVIEW

File: storybook/stories/Examples/AreaChart.stories.tsx [hunk 1]
```diff
@@ -407,3 +407,81 @@ export const AreaChartFillByValue = {
...
+      { month: '2015.08', a: 3490, b: 4300, c: 2100 },
+      { month: '2015.09' },
+    ];
+
+    const toPercent = (decimal: number, fixed = 0) => {
+      return `${(decimal * 100).toFixed(fixed)}%`;
+    };
+
+    return (
+      <AreaChart width={600} height={400} data={data} margin={{ top: 10, right: 30, left: 0, bottom: 0 }}>
```
Comment (ckifer) [inline storybook/stories/Examples/AreaChart.stories.tsx:478]: the y axis on the story gets cut off so I think we need to increase left margin here

**Review by ckifer — COMMENTED** [review]
This is a good fix overall, I know its been an issue for a while.

There is some interesting behavior when only a few areas in the stack have `connectNulls`

With a regular area chart you get a straight line connecting the values where there is null in between such as this story:
image

With a stacked chart you don't get that 
image
The value (yellow here), goes down to either 0 or the lowest point on the stack. 0 is fine if the null are on the start of end of the data and there is nothing to connect to but `connectNulls` is "Whether to connect a graph area across null points." which would draw more like this I think 
image

does `connectNulls` mean go to 0? @nikolasrieble thoughts?

**Comment by nikolasrieble** [comment]
> does `connectNulls` mean go to 0? @nikolasrieble thoughts?

This is a great point, Coltin! The API should behave consistently across components. In this case I would expect to connect between the two points. It should only fall back 0 if not possible (i.e. first or last value)

**Comment by HHongSeungWoo** [comment]
If the `connectNulls` feature is intended to remove noise from the chart, then i agree.

`null null(replace 0) 1 2 null(connect) 3 4 null(replace 0)` Is this the way want it to be processed?

**Comment by HHongSeungWoo** [comment]
We think this needs to be fixed too, right?

image

**Comment by ckifer** [comment]
That's how I would think it would work based off of my current assumptions. 

Maybe we can take a peek at what other charting libs look like, it's kind of hard to know what would be preferable/standard

**Comment by HHongSeungWoo** [comment]
If that's the case, I will revert everything in this PR except for the code cleanup.
It seems that consensus is a priority in addressing this issue

**Comment by ckifer** [comment]
Sounds good, lets revisit this.

One thing is reverting to the old behavior (which is better than current and "fixes" the bugs linked), another is changing the behavior entirely. Maybe we use the old behavior (+connect nulls) for now and later adjust based on what we think/know connectNulls should do

**Comment by ckifer** [comment]
Take this example https://www.highcharts.com/demo/highcharts/area-stacked
and start replacing the data with nulls - without `connectNulls` I think they get behavior correct i.e.
image


With connect nulls they behave the same way that we would have if we merged this fix (go to 0 for all nullish values) 
image

In regular area they connect the line (as `connectNulls` would work for a `Line`) 
image

After looking at that I think how we had it was correct

**Pull request merged**